docs/repo-guides/release-nnf-sw/readme.md

```diff
@@ -0,0 +1,161 @@
+---
+authors: Blake Devcich <blake.devcich@hpe.com>
+categories: release, repo
+---
+
+# Releasing NNF Software
+
+## NNF Software Overview
+
+The following repositories comprise the NNF Software and each have their own versions. There is a
+hierarchy of sorts, since `nnf-deploy` packages the individual components together using submodules.
+
+Each component under `nnf-deploy` needs to be released first, then `nnf-deploy` can be updated to
+point to those release versions, then `nnf-deploy` itself can be updated and released.
+
+The documentation repo (NearNodeFlash/NearNodeFlash.github.io) is released separately and is not
+part of `nnf-deploy`, but it should match the version number of `nnf-deploy`. Release this like the
+other components.
+
+- [NearNodeFlash/nnf-deploy](https://github.com/NearNodeFlash/nnf-deploy)
+    * [NearNodeFlash/nnf-dm](https://github.com/NearNodeFlash/nnf-dm)
+    * [NearNodeFlash/nnf-mfu](https://github.com/NearNodeFlash/nnf-mfu)
+    * [NearNodeFlash/nnf-sos](https://github.com/NearNodeFlash/nnf-sos)
+    * [HewlettPackard/dws](https://github.com/HewlettPackard/dws)
+    * [HewlettPackard/lustre-csi-driver](https://github.com/HewlettPackard/lustre-csi-driver)
+    * [NearNodeFlash/lustre-fs-operator](https://github.com/NearNodeFlash/lustre-fs-operator)
+- [NearNodeFlash/NearNodeFlash.github.io](https://github.com/NearNodeFlash/NearNodeFlash.github.io)
+
+[nnf-ec](https://github.com/NearNodeFlash/nnf-ec) is vendored in as part of `nnf-sos` and does not
+need to be released separately.
+
+## Primer
+
+This document is based on the process set forth by the [DataWorkflowServices Release
+Process](https://dataworkflowservices.github.io/v0.0.1/repo-guides/create-a-release/readme/#github-release-process).
+Please read that as a background for this document before going any further.
+
+## Requirements
+
+To create tags and releases, you will need maintainer or admin rights on the repos.
+
+## Release Each Component In `nnf-deploy`
+
+As mentioned above, you'll first need to create releases for each component contained in
+`nnf-deploy`. This section describes that process.
+
+Each release branch needs to be updated with what is on master. To do that, we'll need the latest
+copy of master, and it will ultimately be merged to the `releases/v0` branch via a Pull Request.
+Once merged, a tag is created and then a release.
+
+<details>
+<summary>Component Version Numbers</summary>
+Each component has its own version number that needs to be incremented. Make sure you change the
```

Comment:
I think this paragraph should not hide.

The v0.0.3 below does not make sense if this paragraph is not expanded, so it doesn't really qualify as a side-note.